base/screens/shop_menu.js

```diff
@@ -0,0 +1,245 @@
+       
+        
+        this.buy_menu = new BuyArtifactsMenu(this.game, this.data, this);
+        this.sell_menu = new SellRepairMenu(this.game, this.data, this);
+
+        this.your_coins_label = this.your_coins_win.set_text_in_position("Your Coins: ", YOUR_COINS_LABEL_X, YOUR_COINS_LABEL_Y);
+        this.your_coins_text = this.your_coins_win.set_text_in_position("", YOUR_COINS_VAL_END_X, YOUR_COINS_VAL_Y, true);
+
+        this.item_name_text = this.item_price_win.set_text_in_position("", ITEM_PRICE_NAME_X, ITEM_PRICE_NAME_Y);
+        this.item_price_label = this.item_price_win.set_text_in_position("Price", ITEM_PRICE_LABEL_X, ITEM_PRICE_LABEL_Y);
+        this.item_price_val_text = this.item_price_win.set_text_in_position("", ITEM_PRICE_VAL_END_X, ITEM_PRICE_VAL_Y, true);
+        this.item_price_coins_label = this.item_price_win.set_text_in_position("Coins", ITEM_PRICE_COINS_X, ITEM_PRICE_COINS_Y);
+
+        this.item_desc_text = this.item_desc_win.set_text_in_position("", ITEM_DESC_TEXT_X, ITEM_DESC_TEXT_Y);
+    }
+
+    set_item_lists(){
+        this.normal_item_list = [];
+        this.artifact_list = [];
+
+        let item_list = this.shops_db[this.shop_key].item_list;
+        for(let i=0; i<item_list.length; i++){
+            let item = this.items_db[item_list[i].key_name];
+            if(item_list[i].quantity === 0) continue;
+
+            if(item.rare_item === true) this.artifact_list.push(item);
+            else this.normal_item_list.push(item);
+        }
+
+        this.normal_item_list = _.mapKeys(this.normal_item_list, item => item.key_name);
+        this.artifact_list = _.mapKeys(this.artifact_list, item => item.key_name);
+    }
+
+    update_your_coins(){
+        this.your_coins_win.update_text(String(this.data.info.party_data.coins), this.your_coins_text);
+    }
+
+    update_item_info(key, custom_price, custom_msg=false, broken=false, cant_sell=false){
+        let this_item = this.data.info.items_list[key];
+
+        this.item_desc_win.update_text(this_item.description, this.item_desc_text);
+        this.item_price_win.update_text(this_item.name, this.item_name_text);
+
+        let coins_label = (custom_msg) ? "" : "Coins";
+        this.item_price_win.update_text(coins_label, this.item_price_coins_label);
+
+        let price_label = "Price";
+        if(custom_msg && !broken) price_label = "It's not broken.";
+        else if(custom_msg && cant_sell) price_label = "We can't buy that.";
+        this.item_price_win.update_text(price_label, this.item_price_label);
+
+        let price_val = custom_price ? custom_price : this_item.price;
+        let price_text = (custom_msg) ? "" : price_val;
+        this.item_price_win.update_text(price_text, this.item_price_val_text);
+    }
+
+    alternate_window_pos(mode){
+        if(this.windows_mode === mode) return;
+        if(mode==="buy"){
```

Comment:
Try to put these key names into variables. At least in consts at the top of the file.